modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/instance/SharedRocksDbInstance.java

```diff
@@ -287,70 +286,95 @@ public ColumnFamily getOrCreateSortedIndexCf(byte[] cfName, int indexId, int tab
     }
 
     /**
-     * Schedules a drop of a column family after destroying an index, if it was the last index managed by that CF.
+     * Removes the given sorted index from this instance. This prevents this index to be returned by {@link #sortedIndexes} call.
      */
-    public CompletableFuture<Void> scheduleIndexCfsDestroy(List<IndexColumnFamily> indexColumnFamilies) {
-        assert !indexColumnFamilies.isEmpty();
+    public void removeSortedIndex(int indexId, ColumnFamily cf) {
+        var cfNameBytes = new ByteArray(cf.nameBytes());
 
-        return flusher.awaitFlush(false)
-                .thenRunAsync(() -> indexColumnFamilies.forEach(this::destroySortedIndexCfIfNeeded), engine.threadPool());
+        sortedIndexCfsByName.computeIfPresent(cfNameBytes, (unused, indexCf) -> {
+            indexCf.indexIdToTableId.remove(indexId);
+
+            return indexCf;
+        });
     }
 
-    void destroySortedIndexCfIfNeeded(IndexColumnFamily indexColumnFamily) {
-        if (!busyLock.enterBusy()) {
-            throw new StorageClosedException();
-        }
+    /**
+     * Schedules a drop of a column family after destroying an index, if it was the last index managed by that CF.
+     */
+    public CompletableFuture<Void> scheduleIndexCfsDestroy(List<ColumnFamily> columnFamilies) {
```

Comment:
This schedules a conditional removal of CFs, but the name does not communicate this. I was confused when I read this call at its call site. Is it possible to make it clear from the method name?

Reply:
Renamed